evalml/tests/preprocessing_tests/test_balanced_classification_data_splitter.py

```diff
@@ -258,3 +258,21 @@ def test_data_splitters_severe_imbalanced_binary_cv():
         y_test = y.iloc[test_indices]
         y_test_counts = y_test.value_counts(normalize=True)
         assert max(y_test_counts.values) == 1140 / 1200
+
+
+@pytest.mark.parametrize("splitter_cls", [BalancedClassificationDataTVSplit, BalancedClassificationDataCVSplit])
+def test_data_splitter_multirun(splitter_cls, X_y_binary, X_y_multi):
```

Comment:
@dsherry I think we also need to test on imbalanced datasets to make sure your fix works as intended. If the dataset is balanced, no random sampling will happen so your change in logic won't actually be used.